datafusion/core/tests/sql/expr.rs

```diff
@@ -491,6 +491,16 @@ async fn test_crypto_expressions() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_array_index() -> Result<()> {
+    test_expression!("([5,4,3,2,1])[1]", "5");
```

Comment:
https://www.postgresql.org/docs/current/arrays.html

I verified that the subscripts are 1 based 👍 

> The array subscript numbers are written within square brackets. By default PostgreSQL uses a one-based numbering convention for arrays, that is, an array of n elements starts with array[1] and ends with array[n].